### DISCUSSION TITLE
expressionof

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#8557] [Language Ideas]
Today you can use the CallerArgumentExpression to get the string equivalent of an argument that is passed to a method. 

There should be an expressionof keyword that does the same thing without the need of a function call. 

Expressionof could be achieved today by creating a method that does the job, but this would actually invoke the expression which is undesirable. Also, a method result is not const.

**Comment by HaloFour** [comment]
What is a use case for wanting to get a string expression for an arbitrary piece of code?
